test/e2e/ownerrefs_finalizers_test.go

```diff
@@ -156,36 +169,68 @@ var (
-		"VSphereClusterIdentity": func(_ types.NamespacedName, owners []metav1.OwnerReference) error {
-			// The vSphereClusterIdentity does not have any owners.
-			return framework.HasExactOwners(owners)
-		},
-		"VSphereDeploymentZone": func(_ types.NamespacedName, owners []metav1.OwnerReference) error {
-			// The vSphereDeploymentZone does not have any owners.
-			return framework.HasExactOwners(owners)
-		},
-		"VSphereFailureDomain": func(_ types.NamespacedName, owners []metav1.OwnerReference) error {
-			// The vSphereFailureDomain can be owned by one or more vSphereDeploymentZones.
-			return framework.HasOneOfExactOwners(owners, []metav1.OwnerReference{vSphereDeploymentZoneOwner}, []metav1.OwnerReference{vSphereDeploymentZoneOwner, vSphereDeploymentZoneOwner})
-		},
+	VSphereReferenceAssertions = func() map[string]func(types.NamespacedName, []metav1.OwnerReference) error {
+		if testMode == SupervisorTestMode {
+			return map[string]func(types.NamespacedName, []metav1.OwnerReference) error{
+				"VSphereCluster": func(_ types.NamespacedName, owners []metav1.OwnerReference) error {
+					return framework.HasExactOwners(owners, clusterController)
+				},
+				"VSphereClusterTemplate": func(_ types.NamespacedName, owners []metav1.OwnerReference) error {
+					return framework.HasExactOwners(owners, clusterClassOwner)
+				},
+				"VSphereMachine": func(_ types.NamespacedName, owners []metav1.OwnerReference) error {
+					return framework.HasExactOwners(owners, machineController)
+				},
+				"VSphereMachineTemplate": func(_ types.NamespacedName, owners []metav1.OwnerReference) error {
+					// The vSphereMachineTemplate can be owned by the Cluster or the ClusterClass.
+					return framework.HasOneOfExactOwners(owners, []metav1.OwnerReference{clusterOwner}, []metav1.OwnerReference{clusterClassOwner})
+				},
+				"VirtualMachine": func(_ types.NamespacedName, owners []metav1.OwnerReference) error {
+					return framework.HasExactOwners(owners, vmwareVSphereMachineController)
+				},
+
+				// Following objects are for vm-operator (not managed by CAPV), so checking ownerReferences is not relevant.
+				"VirtualMachineImage":             func(_ types.NamespacedName, _ []metav1.OwnerReference) error { return nil },
+				"NetworkInterface":                func(_ types.NamespacedName, _ []metav1.OwnerReference) error { return nil },
+				"ContentSourceBinding":            func(_ types.NamespacedName, _ []metav1.OwnerReference) error { return nil },
+				"VirtualMachineSetResourcePolicy": func(_ types.NamespacedName, _ []metav1.OwnerReference) error { return nil },
+				"VirtualMachineClassBinding":      func(_ types.NamespacedName, _ []metav1.OwnerReference) error { return nil },
+				"VirtualMachineClass":             func(_ types.NamespacedName, _ []metav1.OwnerReference) error { return nil },
+				"VMOperatorDependencies":          func(_ types.NamespacedName, _ []metav1.OwnerReference) error { return nil },
+			}
```

Comment:
Q: Do they all still exist after Christians PR to bump to v1alpha2? (just remember that we got rid of a bunch of CRs)

Reply:
I have a follow-up to this PR, that is to check why in the first chance those objects are included in the object graph.
I will review the list as soon as I get to the bottom of this.

Reply:
Thx!